module/zfs/vdev_mirror.c

```diff
@@ -129,9 +130,13 @@ vdev_mirror_load(mirror_map_t *mm, vdev_t *vd, uint64_t zio_offset)
 	 * worse overall when resilvering with compared to without.
 	 */
 
+	/* Fix zio_offset for leaf vdevs */
+	if (vd->vdev_children == 0)
+		zio_offset += VDEV_LABEL_START_SIZE;
+
 	/* Standard load based on pending queue length. */
 	load = vdev_queue_length(vd);
-	lastoffset = vdev_queue_lastoffset(vd);
+	lastoffset = vdev_queue_last_offset(vd);
```

Comment:
nit: If we're going to rename the function to `*_list_offset()` we might as well as the underscore to the local variable too.